src/System.Text.Json/tests/NewtonTests/EnumConverterTests.cs

```diff
@@ -0,0 +1,532 @@
+        {
+            NegativeFlagsEnumClass negativeEnumClass = new NegativeFlagsEnumClass();
+            negativeEnumClass.Value1 = NegativeFlagsEnum.NegativeFour | NegativeFlagsEnum.NegativeTwo;
+            negativeEnumClass.Value2 = NegativeFlagsEnum.Two | NegativeFlagsEnum.Four;
+
+            string json = JsonSerializer.ToString(negativeEnumClass, new JsonSerializerOptions { WriteIndented = true });
+
+            Assert.Equal(@"{
+  ""Value1"": -2,
+  ""Value2"": 6
+}", json);
+        }
+
+        [Fact]
+        public static void SerializeNegativeEnum()
+        {
+            NegativeEnumClass negativeEnumClass = new NegativeEnumClass()
+            {
+                Value1 = NegativeEnum.Negative,
+                Value2 = (NegativeEnum)int.MinValue
+            };
+
+            string json = JsonSerializer.ToString(negativeEnumClass, new JsonSerializerOptions { WriteIndented = true });
+
+            Assert.Equal(@"{
+  ""Value1"": -1,
+  ""Value2"": -2147483648
+}", json);
+        }
+
+        [Fact]
+        public static void EnumMemberPlusFlags()
+        {
+            List<Foo> lfoo =
+                new List<Foo>
+                {
+                    Foo.Bat | Foo.SerializeAsBaz,
+                    Foo.FooBar,
+                    Foo.Bat,
+                    Foo.SerializeAsBaz,
+                    Foo.FooBar | Foo.SerializeAsBaz,
+                    (Foo)int.MaxValue
+                };
+
+            string json1 = JsonSerializer.ToString(lfoo, new JsonSerializerOptions { WriteIndented = true, PropertyNamingPolicy = JsonNamingPolicy.CamelCase });
+
+            Assert.Equal(@"[
+  6,
+  1,
+  2,
+  4,
+  5,
+  2147483647
+]", json1);
+
+            IList<Foo> foos = JsonSerializer.Parse<List<Foo>>(json1);
+
+            Assert.Equal(6, foos.Count);
+            Assert.Equal(Foo.Bat | Foo.SerializeAsBaz, foos[0]);
+            Assert.Equal(Foo.FooBar, foos[1]);
+            Assert.Equal(Foo.Bat, foos[2]);
+            Assert.Equal(Foo.SerializeAsBaz, foos[3]);
+            Assert.Equal(Foo.FooBar | Foo.SerializeAsBaz, foos[4]);
+            Assert.Equal((Foo)int.MaxValue, foos[5]);
+
+            List<Bar> lbar = new List<Bar>() { Bar.FooBar, Bar.Bat, Bar.SerializeAsBaz };
+
+            string json2 = JsonSerializer.ToString(lbar, new JsonSerializerOptions { WriteIndented = true, PropertyNamingPolicy = JsonNamingPolicy.CamelCase });
+
+            Assert.Equal(@"[
+  0,
+  1,
+  2
+]", json2);
+
+            IList<Bar> bars = JsonSerializer.Parse<List<Bar>>(json2);
+
+            Assert.Equal(3, bars.Count);
+            Assert.Equal(Bar.FooBar, bars[0]);
+            Assert.Equal(Bar.Bat, bars[1]);
+            Assert.Equal(Bar.SerializeAsBaz, bars[2]);
+        }
+
+        [Fact]
+        public static void DataContractSerializerDuplicateNameEnumTest()
+        {
+            MemoryStream ms = new MemoryStream();
+            var s = new DataContractSerializer(typeof(DuplicateEnumNameTestClass));
```

Comment:
It doesn't make sense to add `DataContractSerializer` tests here. I would remove them, unless they were somehow necessary for testing the `JsonSerializer`.

Reply:
I guess it is not officially supported, test passed for serializer but not for deserializer. So thought it will not hurt to have it, but sure if you guys think don't need this will remove

Reply:
> I guess it is not officially supported, test passed for serializer but not for deserializer. 

What do you mean they pass for serializer but not for deserializer?

The goal here is to add tests for the `System.Text.Json.Serializer` type (both read/writer). This test case, as far as I can tell, isn't doing that. It is instead testing the `DataContractSerializer` type.

Reply:
removed